### PR TITLE
Deep url

### PR DESCRIPTION
**Pull request by SoIidarity**

### CODE REVIEW

File: src/components/pages/CardView.vue [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  },
+  computed: {
+    ...mapState(["deepUrlCard"])
+  },
+  mounted() {
+    console.log("loading Started");
+    let currentPath = this.$route.path;
+    let tokenId = currentPath.substring(
+      currentPath.lastIndexOf(":") + 1,
+      currentPath.length
```
Comment (jamesmorgan) [inline src/components/pages/CardView.vue]: Im not sure this is needed? In vue you are call `this.$route.params.id` if you change you route path to be `/card/:id`? Maybe I am missing what you are trying to achieve here.

File: src/components/pages/CardView.vue [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    );
+    this.$store.dispatch(actions.LOAD_DEEP_URL_CARD, { tokenId });
+    console.log("mounted");
+  },
+  methods: {
+    loadIntoStoreCard() {
```
Comment (jamesmorgan) [inline src/components/pages/CardView.vue]: What triggers this method?

File: src/router/index.js [hunk 3]
```diff
@@ -30,6 +31,11 @@ export default new Router({
       path: '/account',
       name: 'account',
       component: Account
+    },
+    {
+      path: '/viewCard:id',
```
Comment (jamesmorgan) [inline src/router/index.js]: Should this not be `/card/:id` or similar?

File: src/store/index.js [hunk 4]
```diff
@@ -243,6 +249,53 @@ const store = new Vuex.Store({
...
+          let extra = accountToken[3]
+          let cardIndex = accountToken[4]
+          let benefactorIndex = accountToken[5].toNumber()
+          if (state.cards) {
+            let cardInformation = state.cards.filter(card => {
+              return card.cardIndex === cardIndex.toNumber();
+            });
+            let accountCreatedCard = false
+            if (state.account) {
+              accountCreatedCard = (state.account.toLowerCase() === gifter.toLowerCase()) //if the current account created the card
```
Comment (jamesmorgan) [inline src/store/index.js]: Comparing accounts works like this but we have found that if you checksum the accounts it a little less clunky and you can then set them in the store checksummed in the first place, you don't need to lowercase to equality check them `https://web3js.readthedocs.io/en/1.0/web3-utils.html#tochecksumaddress`

File: src/store/index.js [hunk 5]
```diff
@@ -243,6 +249,53 @@ const store = new Vuex.Store({
...
+    }, {
+      tokenId
+    }) {
+      if (state.deepUrlCardNumber === null) {
+        commit(mutations.SET_DEEP_URL_CARD_NUMBER, tokenId)
+
+      } else {
+        if (state.contract) {
+          const contract = await state.contract.deployed();
+          let accountToken = await contract.tokenDetails(tokenId);
```
Comment (jamesmorgan) [inline src/store/index.js]: Would be nice to wrap this logic in a mapping method that returns an object in the right state/types etc.

File: src/components/pages/CardView.vue (hunk 1 above)
Comment (SoIidarity) [reply src/components/pages/CardView.vue]: I didnt know about `this.$route.params.id` much better solution to what I was doing! updated the component and the router to achieve your implementation.

File: src/components/pages/CardView.vue (hunk 2 above)
Comment (SoIidarity) [reply src/components/pages/CardView.vue]: nothing. removed and cleaned up the surrounding code.

File: src/router/index.js (hunk 3 above)
Comment (SoIidarity) [reply src/router/index.js]: indeed it should. have updated it in accordance with the `this.$route.params.id` update

File: src/store/index.js (hunk 4 above)
Comment (SoIidarity) [reply src/store/index.js]: cool. thanks for the recommendation. implemented this

File: src/store/index.js (hunk 5 above)
Comment (SoIidarity) [reply src/store/index.js]: I agree. I think we can move this to the next iteration though as this works fine for now? Lets release then can do a big refactor. there are a number of other places that require this kind of refactor within the store.

**Pull request merged**